### PR TITLE
EDSC-2910: Toggle Granule Filter Keyboard Shortcut

### PR DESCRIPTION
**Pull request by bensun13**

### CODE REVIEW

**Comment by mreese84** [comment]
@bensun13 this looks pretty good. We noticed that we did not have a unite test for the existing keyboard shortcut, so we added one with [this PR](https://github.com/nasa/earthdata-search/pull/1232). If you can get a test added we'll do a full review of your PR. Thank you!

**Review by abbottry — CHANGES_REQUESTED** [review]
We noticed one issue, when not on the granules page, pressing `g` still toggles this redux state resulting in the panel being open when you eventually visit this page. We have a utility that you can use to check this, `pathStartsWith()` -- please use this to ensure that the user is looking at the correct page using `pathStartsWith(location.pathname, ['/search/granules']). `location.pathname` should be available to you in redux.

Additionally, please add a unit test.

**Comment by bensun13** [comment]
> We noticed one issue, when not on the granules page, pressing `g` still toggles this redux state resulting in the panel being open when you eventually visit this page. We have a utility that you can use to check this, `pathStartsWith()` -- please use this to ensure that the user is looking at the correct page using `pathStartsWith(location.pathname, ['/search/granules']). `location.pathname` should be available to you in redux.
> 
> Additionally, please add a unit test.

Added unit test and check for path

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsHeader.test.js [hunk 1]
```diff
@@ -295,4 +307,33 @@ describe('granuleFilters link', () => {
       expect(props.onToggleSecondaryOverlayPanel).toHaveBeenCalledWith(false)
     })
   })
+
+  describe('when the g key is pressed', () => {
+    test('toggles the granule panel state correctly', () => {
```
Comment (macrouch) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsHeader.test.js:312]: Can you add one more test that shows `onToggleSecondaryOverlayPanel` does not get called with a different `location.pathname`, `/search` for example

Comment (bensun13) [reply static/src/js/components/GranuleResults/__tests__/GranuleResultsHeader.test.js:312]: Added test for pathname.

**Review by macrouch — APPROVED** [review]

**Review by abbottry — APPROVED** [review]

**Comment by mreese84** [comment]
@bensun13 thank you for contributing! This feature will be deployed to our production environment no later than 10/28/20 as part of our 1.130.x release.

[search.earthdata.nasa.gov](https://search.earthdata.nasa.gov)

**Comment by mreese84** [comment]
Fixes #1221

**Pull request merged**